activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/variable/JsonType.java

```diff
@@ -50,40 +49,25 @@ public boolean isCachable() {
   }
 
   public Object getValue(ValueFields valueFields) {
-    Object jsonValue = null;
+    Object loadedValue = null;
     if (valueFields.getTextValue() != null && valueFields.getTextValue().length() > 0) {
-      if(jsonValue==null) {
         try {
-          jsonValue = objectMapper.readTree(valueFields.getTextValue());
+            loadedValue = jsonTypeConverter.convertToValue(
+                objectMapper.readTree(valueFields.getTextValue()), valueFields);
 
         } catch (Exception e) {
           logger.error("Error reading json variable " + valueFields.getName(), e);
         }
       }
-      if(jsonValue!=null && StringUtils.isNotBlank(javaClassFieldForJackson) ) {
-        //can find type so long as JsonTypeInfo annotation on the class - see https://stackoverflow.com/a/28384407/9705485
-        JsonNode classNode = ((JsonNode)jsonValue).get(javaClassFieldForJackson);
-        if(classNode != null) {
-          final String type = classNode.asText();
-          Class<?> cls = null;
-          try {
-            cls = Class.forName(type, false, this.getClass().getClassLoader());
-          } catch (ClassNotFoundException e) {
-            logger.warn("Unable to obtain type for json variable object " + valueFields.getName(), e);
-          }
-          if(cls!=null) {
-            jsonValue = objectMapper.convertValue(jsonValue, cls);
-          }
-        }
-      }
-    }
-    return jsonValue;
+    return loadedValue;
   }
 
-
   public void setValue(Object value, ValueFields valueFields) {
     try {
       valueFields.setTextValue(objectMapper.writeValueAsString(value));
+      if (value != null) {
+          valueFields.setTextValue2(value.getClass().getName());
```

Comment:
Here, I'm reusing a field that already exists but that's not used by JSON type (`textValue2`) to store the value of the original Java Class. Is it ok or is it better to create a new column in the database base (`javaType`) for that?

Reply:
I'll create a specific field: it will be cleaner.

Reply:
I agree

Reply:
FYI, for JPA variables types textValue and textValue2 were used on for the value and the other for the class name. 
https://github.com/Activiti/Activiti/blob/develop/activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/variable/JPAEntityVariableType.java#L67

Reply:
Thanks, @balsarori. Yes, I reached that code as well when I was trying to understand what was the purpose of `textValue2`. It's just hard to know who is who, with such generic names. That's why I was thinking about creating an `originalClassName` attribute.

Reply:
For simplicity, I'll use `textValue2` without adding a new field.